lib/trino-filesystem/src/main/java/io/trino/filesystem/hdfs/HdfsFileSystem.java

```diff
@@ -125,4 +132,29 @@ public FileIO toFileIo()
     {
         return new ForwardingFileIo(this);
     }
+
+    @Override
+    public void batchDeleteFiles(Collection<String> paths)
+            throws IOException
+    {
+        if (!paths.isEmpty()) {
```

Comment:
I went through the `hadoop-apache` library and it seems that the components of the path which are used in the initialisation of the `FileSystem` instance are:
- scheme
- authority


See `org.apache.hadoop.fs.Path#getFileSystem` , `org.apache.hadoop.fs.FileSystem#get(java.net.URI, org.apache.hadoop.conf.Configuration)`

```
    String scheme = uri.getScheme();
    String authority = uri.getAuthority();
```

I don't know whether the grouping per directory is actually necessary.

Reply:
Probably it is not. @findepi as this was your idea WDYT ?

Reply:
> I went through the `hadoop-apache` library and it seems that the components of the path which are used in the initialisation of the `FileSystem` instance are:
> 
> * scheme
> * authority

Are we talking about contract/interface or implementation?

Also, does https://trino.io/docs/current/connector/hive-s3.html#s3-security-mapping > "prefix" matter?

Reply:
I missed the fact that the path can be used to add, based on the `Path`  dynamic configuration settings while creating the `FileSystem` implementation (see `DynamicConfigurationProvider`) .
So quite likely the whole path is needed.

See s3 predicate matching:

https://github.com/trinodb/trino/blob/fe608f2723842037ff620d612a706900e79c52c8/plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/S3SecurityMapping.java#L158-L165

Reply:
if we were to fully honor it, we would need to poke into dynamic configuration and see whether all paths in the group would yield the same configs. It's possible (Trino has all the information required), but hard (the information is well interfaced-out and inaccessible). I don't think there is any reasonable real-life use-case to have different configuration for two snapshot files of a table. That's why i deemed grouping by directory should work well in practice.

cc @losipiuk @arhimondr @electrum